tftb/generators/misc.py

```diff
@@ -56,9 +56,17 @@ def atoms(n_points, coordinates):
     """
     # FIXME: This function produces incorrect output when coordinates are
     # one-dimensional.
+    # yoder: would the fix be something like: if not isinstance(coordinates[0], list): coordinates = [coordinates]
+    # ???
+    # at first glance, it looks ok. let's just add some data typ handlers. they may be imperfect, but the basic idea is to catch
+    # some common errors, like passing a list, not an array.
+    # an un-contained 1-d coordinates array:
+    if not hasattr(coordinates[0], '__len__'): coordinates = [coordinates]
+    if not hasattr(coordinates, 'shape'):      coordinates = numpy.array(coordinates)
+    #
```

Comment:
agreed. this can be fixed later, but i recommend leaving some comments to guide that process.